src/Models/Product.php

```diff
@@ -164,92 +164,114 @@ public function scopeByIds(Builder $query, array $productIds): Builder
-            }
+                    if (isset($child->special_from_date) && $child->special_from_date > now()->toDateTimeString()) {
+                        return false;
+                    }
 
-            if (isset($child->special_to_date) && $child->special_to_date < now()->toDateTimeString()) {
-                return false;
-            }
+                    if (isset($child->special_to_date) && $child->special_to_date < now()->toDateTimeString()) {
+                        return false;
+                    }
 
-            return true;
-        })->min->special_price;
+                    return true;
+                })->min->special_price;
+            }
+        );
     }
 
-    public function getMinSaleQtyAttribute(int $minSalesQty): int
+    public function minSaleQty(): Attribute
     {
-        if (! $this->qty_increments) {
-            return $minSalesQty;
-        }
-        $remainder = $minSalesQty % $this->qty_increments;
-        if ($remainder === 0) {
-            return $minSalesQty;
-        }
+        return Attribute::make(
+            get: function (?int $minSaleQty): ?int {
+                if (! $this->qty_increments) {
+                    return $minSaleQty;
+                }
+                $remainder = $minSaleQty % $this->qty_increments;
+                if ($remainder === 0) {
+                    return $minSaleQty;
+                }
 
-        return $minSalesQty - $remainder + $this->qty_increments;
+                return $minSaleQty - $remainder + $this->qty_increments;
+            }
+        );
     }
 
-    public function getUrlAttribute(): string
+    public function url(): Attribute
     {
-        return '/' . ($this->url_key ? $this->url_key . Rapidez::config('catalog/seo/product_url_suffix') : 'catalog/product/view/id/' . $this->entity_id);
+        return Attribute::make(
+            get: fn (): string => '/' . ($this->url_key ? $this->url_key . Rapidez::config('catalog/seo/product_url_suffix') : 'catalog/product/view/id/' . $this->entity_id)
+        );
     }
 
-    public function getImagesAttribute(): array
+    public function images(): Attribute
     {
-        return $this->gallery->sortBy('productImageValue.position')->pluck('value')->toArray();
+        return Attribute::make(
+            get: fn (): array => $this->gallery->sortBy('productImageValue.position')->pluck('value')->toArray()
+        );
     }
 
-    public function getImageAttribute($image): ?string
+    public function image(): Attribute
     {
-        return $image !== 'no_selection' ? $image : null;
+        return Attribute::make(
+            get: fn (?string $image): ?string => $image !== 'no_selection' ? $image : null
+        );
     }
 
-    public function getSmallImageAttribute($image): ?string
+    public function smallImage(): Attribute
     {
-        return $this->getImageAttribute($image);
+        return Attribute::make(
+            get: fn (): ?string => $this->image
```

Comment:
Wouldn't we be missing the ability to pass `$image` through to the image attribute?